cf-testd/cf-testd.c

```diff
@@ -0,0 +1,492 @@
+#include <crypto.h>                     // CryptoInitialize
+
+#include <cf-serverd-enterprise-stubs.h>
+#include <net.h>
+#include <server_common.h>
+#include <server_access.h>              // acl_Free
+#include <item_lib.h>                   // DeleteItemList
+#include <server_transform.h>           // Summarize
+#include <bootstrap.h>                  // GetAmPolicyHub
+#include <policy_server.h>              // PolicyServerReadFile
+#include <systype.h>                    // CLASSTEXT
+#include <mutex.h>                      // ThreadLock
+#include <locks.h>                      // AcquireLock
+#include <exec_tools.h>                 // ActAsDaemon
+#include <man.h>                        // ManPageWrite
+#include <server_tls.h>                 // ServerTLSInitialize
+#include <tls_generic.h>                // TLSLogError
+#include <timeout.h>                    // SetReferenceTime
+#include <known_dirs.h>                 // GetInputDir
+#include <sysinfo.h>                    // DetectEnvironment
+#include <time_classes.h>               // UpdateTimeClasses
+#include <loading.h>                    // LoadPolicy
+#include <printsize.h>                  // PRINTSIZE
+#include <conversion.h>                 // MapAddress
+#include <server_code.h>                // InitServer
+#include <signals.h>                    // ReloadConfigRequested
+#include <openssl/err.h>                // ERR_get_error
+
+typedef int (*InitServerFunction)(size_t queue_size);
+
+// ============================= CFTestD_Config ==============================
+typedef struct {
+    char *file;
+}CFTestD_Config;
+
+/*******************************************************************/
+/* Command line option parsing                                     */
+/*******************************************************************/
+
+static const char *const CF_TESTD_SHORT_DESCRIPTION = "CFEngine dummy server";
+
+static const char *const CF_TESTD_MANPAGE_LONG_DESCRIPTION =
+        "Used for testing CFEngine by mimicing cf-serverd.";
+
+static const struct option OPTIONS[] =
+{
+    {"debug",     no_argument,       0, 'd'},
+    {"file",      required_argument, 0, 'f'},
+    {"help",      no_argument,       0, 'h'},
+    {"inform",    no_argument,       0, 'I'},
+    {"timestamp", no_argument,       0, 'l'},
+    {"verbose",   no_argument,       0, 'v'},
+    {"version",   no_argument,       0, 'V'},
+    {NULL,        0,                 0, '\0'}
+};
+
+static const char *const HINTS[] =
+{
+    "Enable debugging output",
+    "Read report from file",
+    "Print the help message",
+    "Print basic information about what cf-testd does",
+    "Log timestamps on each line of log output",
+    "Output verbose information about the behaviour of the agent",
+    "Output the version of the software",
+    NULL
+};
+
+CFTestD_Config *CFTestD_ConfigInit()
+{
+    CFTestD_Config* r = (CFTestD_Config *)(xmalloc(sizeof(CFTestD_Config)));
+    r->file = NULL;
```

Comment:
``xcalloc()`` can be used here making the ``r->file`` assignment unnecessary.

Reply:
Thanks, fixed.